src/ray/util/event.h

```diff
@@ -164,6 +172,12 @@ class RayEvent {
     return *this;
   }
 
+  template <typename T>
+  RayEvent &WithField(const std::string &key, const T &value) {
+    custom_fields_[key] = value;
```

Comment:
What's going to happen if we try calling the unsupported types for json?

Reply:
The compiling will fail. Like this 
```
external/nlohmann_json/include/nlohmann/json.hpp:2174:17: note:   no known conversion for argument 1 from ‘const ray::test_cls’ to ‘nlohmann::basic_json<>’
```